src/main/java/org/opensearch/knn/index/query/KNNScorer.java

```diff
@@ -87,6 +87,13 @@ public float score() throws IOException {
             public int docID() {
                 return docIdsIter.docID();
             }
+
+            @Override
+            public boolean equals(Object obj) {
```

Comment:
qq: Why add?

Reply:
- Instead of returning null I am returning [empty map](https://github.com/opensearch-project/k-NN/pull/1877/files#diff-b0eb527e944c912b803db37ed9dd3dfa7689864b77d7146a0d45d7b73138d7b8R125)
- as a consequence I had to [change the unit test](https://github.com/opensearch-project/k-NN/pull/1877/files#diff-1cd3834579e8736ab1ebfec9ffeaef363d96f76443ba40d166ed368cf6c6cedaR353)

`assertEquals` uses equals method to match objects. Comparing objects directly makes for a tight test